UnityProject/Assets/Scripts/UI/UI_OxygenAlert.cs

```diff
@@ -10,12 +10,36 @@ public class UI_OxygenAlert : MonoBehaviour {
 
 	public Image img;
 	private Sprite sprite;
+	private float timeWait;
 
-	void Start ()
+	void Start()
 	{
 		img = GetComponent<Image>();
 		sprite = img.sprite;
-		InvokeRepeating("CycleImg", 1f, 1f); //Cycle images every 1 second
+	}
+
+	private void OnEnable()
+	{
+		UpdateManager.Instance.Add(UpdateMe);
+	}
+
+	private void OnDisable()
+	{
+		if (UpdateManager.Instance != null)
+		{
+			UpdateManager.Instance.Remove(UpdateMe);
+		}
+	}
+
+	void UpdateMe()
+	{
+		timeWait += Time.deltaTime;
+		// Cycle images every 1 second
+		if (timeWait > 1f)
+		{
+			CycleImg();
+			timeWait = 0f;
```

Comment:
This would technically only be accurate with infinite FPS, because when we have, say 400ms per frame it would
```
timeWait = 0;
timeWait = 0.4;
timeWait = 0.8;
timeWait = 1.2 -> 0 // Cycle image called;
```
This means that that the lower the FPS the more inaccurate the cycle time is.
Should be changed to `timeWait -= CycleTime;`

Reply:
I've updated the code to subtract the new constant instead of setting timeWait to 0f.